holo-ospf/src/events.rs

```diff
@@ -381,9 +383,63 @@ where
     // Examine LLS data block if enabled and present in the packet.
     if iface.config.lls_enabled
         && hello.options().l_bit()
-        && let Some(_lls) = hello.lls()
+        && let Some(lls) = hello.lls()
     {
-        // TODO: Handle LLS data
+        // Handle Reverse Metric (RFC 9339).
+        if iface.config.reverse_metric.receive {
+            // We have to recall the latest configuration we have received.
+            // If it has not changed, we do not update the metric nor we
+            // originate a new Router LSA.
+            // If the Reverse Metric from the neighbor has been updated since
+            // the last one we received, then we propagate the information.
+            for (mtid, (flags, metric)) in lls.reverse_metric.iter() {
+                let entry = iface.config.reverse_metric.rm_rx.entry(*mtid);
+                let higher = flags.contains(ReverseMetricFlags::H);
+                let offset = flags.contains(ReverseMetricFlags::O);
+                if let Some(cfg) = match entry {
+                    Entry::Occupied(entry) => {
+                        // Has the RM changed?
+                        let old = entry.into_mut();
+                        (&old.metric != metric
+                            || old.higher != higher
+                            || old.offset != offset)
+                            .then(|| {
+                                old.metric = *metric;
+                                old.higher = higher;
+                                old.offset = offset;
+                                old
+                            })
+                    }
+                    Entry::Vacant(entry) => {
+                        // This is the first time we see the RM.
+                        Some(entry.insert(ReverseMetricMtCfg {
+                            metric: *metric,
+                            higher,
+                            offset,
+                        }))
+                    }
+                } {
+                    // Update metric.
+                    if cfg.higher && iface.config.cost < cfg.metric
+                        || !cfg.higher
+                    {
+                        if cfg.offset {
+                            iface.config.cost =
+                                iface.config.cost.saturating_add(cfg.metric);
+                        } else {
+                            iface.config.cost = cfg.metric;
+                        }
```

Comment:
This is not correct. We can not change the cost loaded from the configuration. We should rather extend the `InterfaceState` with the current cost value so that we can fallback later on the provisioned cost.

Reply:
Fixed in https://github.com/holo-routing/holo/commit/46ab06ea67338b611f1cd149eb64b924931f16ce.